### PR TITLE
Accept required and requisite control flag for pam_pwhistory

### PR DESCRIPTION
**Pull request by marcusburghardt**
#### Description:

In RHEL8.8 and RHEL9.2 it was introduced a new `authselect` feature to  enable `pam_pwhistory` module.
Before the feature, the module was enabled by creating a custom `authselect` profile.
Historically, the control flag used to enable `pam_pwhistory.so` was `required`.
However, the `authselect` feature uses `requisite` instead.
These two control flags are correct and the differences are only about the module reporting, without affecting its effect.
Therefore, the assessment should pass if either `required` or `requisite` is used.

#### Rationale:

Thanks to @vojtapolasek researches, it was noticed a possible scenario where the feature is introduced in a system with a custom profile based in older `authselect` versions. In these cases, the system would be compliant by using the `required` control flag. However, since the rule is expecting only `requisite`, it will fail and the remediation would be executed.
However, in these cases, the remediation can't be safely ensured because the remediation will try to use the feature but the custom profile is not aware of the feature. The custom profile should be updated or recreated by the administrator.

#### Review Hints:

This is a corner case difficult to be tested.
Basically, it would be necessary to scan and remediate a system with the `authselect` package in a old version, like in the current RHEL8.7.
Then, the system should be updated to use the new `authselect` package version, which includes the `with-pwhistory` feature.

Before this patch, the scan with the new `authselect` package should fail because the control flag is set to `required`. Consequently, the remediation will fail due to the above mentioned issue.

After this patch, the scan with the new `authselect` package will pass because both `required` and `requisite` are accepted. No remediation will be necessary.

### CODE REVIEW

**Comment by marcusburghardt** [comment]
The #10021 is also related to this.

File: linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/var_password_pam_remember_control_flag.var [hunk 1]
```diff
@@ -20,4 +20,6 @@ options:
     "sufficient": "sufficient"
     "binding": "binding"
     "ol8": "required,requisite"
+    "rhel8": "requisite,required"
```
Comment (yuumasato) [inline linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/var_password_pam_remember_control_flag.var:23]: I would rather have a reusable option:
```suggestion
    "requisite_and_requried": "requisite,required"
```

And then other products could use it as well.

File: linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_password_pam_pwhistory_remember_password_auth/rule.yml [hunk 2]
```diff
@@ -129,3 +129,7 @@ warnings:
        Newer versions of <tt>authselect</tt> contain an authselect feature to easily and properly
        enable <tt>pam_pwhistory.so</tt> module. If this feature is not yet available in your
        system, an authselect custom profile must be used to avoid integrity issues in PAM files.
+       If a custom profile was created and used in the system before this authselect feature be
+       available, the new feature can't be used with the outdated custom profile and the
+       remediation will fail. In this case, the custom profile should be recreated or manually
+       updated.
```
Comment (yuumasato) [inline linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_password_pam_pwhistory_remember_password_auth/rule.yml:135]:
```suggestion
       If a custom profile was created and used in the system before this authselect feature was
       available, the new feature can't be used with this custom profile and the
       remediation will fail. In this case, the custom profile should be recreated or manually
       updated.
```

**Review by yuumasato — CHANGES_REQUESTED** [review]
@marcusburghardt I'm okay leaving the consolidation of the variables to unique selector to a future PR.

File: linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/var_password_pam_remember_control_flag.var (hunk 1 above)
Comment (marcusburghardt) [reply linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/var_password_pam_remember_control_flag.var:23]: It is a good suggestion. I like more generic variables and also it is easier to understand the intention when reading `requisite_or_required` than `rhel8`, for example. I am updating.

Comment (marcusburghardt) [reply linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/var_password_pam_remember_control_flag.var]: Done.

**Review by yuumasato — APPROVED** [review]
LGTM, waiting for tests by @vojtapolasek

**Review by vojtapolasek — CHANGES_REQUESTED** [review]
Hello @marcusburghardt  and thanks for this fix. I have tested it on 8.7 -> 8.8 as well as on 9.1 -> 9.2 and it works nicely.
Just a small note, see the comment and it is ready to be merged.

File: tests/data/profile_stability/rhel8/stig.profile [hunk 3]
```diff
@@ -363,6 +363,7 @@ selections:
 - set_password_hashing_algorithm_passwordauth
 - set_password_hashing_algorithm_systemauth
 - set_password_hashing_min_rounds_logindefs
+- ssh_keys_passphrase_protected
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply tests/data/profile_stability/rhel8/stig.profile:366]: It was probably a change in some script used in the build process, outside the scope of this PR. I noticed that this line was simply repositioned alphabetically.

[the remark this answers is not in the document]
Comment (yuumasato) [reply tests/data/profile_stability/rhel8/stig.profile:366]: Good catch Vojta.
I see the new rule was added by this PR https://github.com/ComplianceAsCode/content/pull/10017, but it did not add the rule to stabilization.

@marcusburghardt Could you make another PR for that?

[the remark this answers is not in the document]
Comment (marcusburghardt) [reply tests/data/profile_stability/rhel8/stig.profile:366]: I can, but we would need to review and merge it before this one. Because it will need to be rebased.
@yuumasato , this line is not new in the files. It is only in a different position.

[the remark this answers is not in the document]
Comment (vojtapolasek) [reply tests/data/profile_stability/rhel8/stig.profile:366]: I mean, if you just remove those changes related to ssh_rule from this PR (so that changes are related only to the PAM flag), nothing should break. Or am I missing something?

[the remark this answers is not in the document]
Comment (marcusburghardt) [reply tests/data/profile_stability/rhel8/stig.profile:366]: But the baseline for the stability test will still be broken. It needs to be updated anyways. Because that I said it should be fixed then this PR should be rebased if we want to do it in a separate PR. But reinforcing, the change is simply about the line position, nothing added or removed around the ssh line. What is the issue of keeping it? I think the effort to separate this in a different PR and rebasing this one is not worth unless we have a real issue with that.

[the remark this answers is not in the document]
Comment (yuumasato) [reply tests/data/profile_stability/rhel8/stig.profile:366]: I don't understand why a rebase would be required.

[the remark this answers is not in the document]
Comment (marcusburghardt) [reply tests/data/profile_stability/rhel8/stig.profile]: I agreed to manually update the baseline for the STIG stability checks in the last commit in order to help @yuumasato to accelerate some offline tests based on a offline conversation we had. However, the #10182 must me merged as soon as possible to keep the STIG stability checks consistent in Upstream.

**Comment by vojtapolasek** [comment]
@marcusburghardt Please note that the profile stability test was passing even before you shuffled the ssh_keys_passphrase_protected lines in the /tests/data/profile_stability/rhel8/stig.profile. The profile stability test is based on set comparison, the order of lines does not matter. I have tested it - I have checked out the master branch before your PR and the profile stability test for RHEL 8 was passing.
Therefore, I am sure that the change of lines involving a/tests/data/profile_stability/rhel8/stig.profile is not in scope of this PR and actually does not have any considerable impact at all. I see you proposed the change in a separate PR https://github.com/ComplianceAsCode/content/pull/10182 . That is nice, but this PR does not have to wait for that PR to be merged. Please remove changes WRT ssh_keys_passphrase_protected from this PR so that it can be treated as a single unit.

**Comment by marcusburghardt** [comment]
> @marcusburghardt Please note that the profile stability test was passing even before you shuffled the ssh_keys_passphrase_protected lines in the /tests/data/profile_stability/rhel8/stig.profile. The profile stability test is based on set comparison, the order of lines does not matter. I have tested it - I have checked out the master branch before your PR and the profile stability test for RHEL 8 was passing. Therefore, I am sure that the change of lines involving a/tests/data/profile_stability/rhel8/stig.profile is not in scope of this PR and actually does not have any considerable impact at all. I see you proposed the change in a separate PR #10182 . That is nice, but this PR does not have to wait for that PR to be merged. Please remove changes WRT ssh_keys_passphrase_protected from this PR so that it can be treated as a single unit.

Yes @vojtapolasek , I am aware that this ordering does not affect the result of the test, but I still can't be complacent with the issue that caused the line to be reordered in this PR. We should rely on the `stig.profile` and `stig_gui.profile` which are generated automatically and ideally should never be necessary to do manual changes like this on them.

I believe you missed my last comment in the thread, where I informed that the last commit was already updated and the condition on which I agreed to make this manual change in this PR. In any case, the source of issue will now be resolved by #10182, which should be merged as soon as possible to avoid possible "extra updates" in future PRs related to changes in STIG profiles.

**Comment by vojtapolasek** [comment]
Great. I will merge it as soon as all tests pass.a

**Review by vojtapolasek — APPROVED** [review]
